apps/web/src/components/bank-account-form/sections/bank-account-form.tsx

```diff
@@ -0,0 +1,244 @@
+                      amount: formatCurrency(initialBid, lang),
+                    })
+                  : undefined
+              }
+              id="bid-input"
+              intlConfig={{ locale, currency: DEFAULT_CURRENCY }}
+              label={t('forms:fields.bid.label')}
+              name="bid-input"
+              value={bid || ''}
+              variant={'small'}
+            />
+            {/* Force formData to be built from this "unmasked" value */}
+            <input id="bid" name="bid" type="hidden" value={bid as string} />
+            <Checkbox
+              checked={isConfirmed}
+              id="confirmBid"
+              name="confirmBid"
+              label={t('forms:fields.bid.confirmation')}
+              onChange={() => setIsConfirmed(!isConfirmed)}
+            />
+          </>
+        ) : (
+          <>
+            <Heading level={2}>{t('forms:sections.Bank Account')}</Heading>
+
+            <TextInput
+              error={
+                formErrors && 'accountNumber' in formErrors
+                  ? (formErrors.accountNumber as string)
+                  : ''
+              }
+              label={t('forms:fields.accountNumber.label')}
+              name="accountNumber"
+              variant="small"
+            />
+
+            <TextInput
+              error={
+                formErrors && 'routingNumber' in formErrors
+                  ? (formErrors.routingNumber as string)
+                  : ''
+              }
+              label={t('forms:fields.routingNumber.label')}
+              name="routingNumber"
+              variant="small"
+            />
+
+            <TextInput
+              error={
+                formErrors && 'fullName' in formErrors
+                  ? (formErrors.fullName as string)
+                  : ''
+              }
+              helpText={t('forms:fields.fullName.helpText')}
+              label={t('forms:fields.fullName.label')}
+              name="fullName"
+              placeholder="Jane Smith"
+              variant="small"
+            />
+
+            <BillingAddress
+              formErrors={{
+                address1:
+                  formErrors && 'address1' in formErrors
+                    ? (formErrors.address1 as string)
+                    : '',
+                city:
+                  formErrors && 'city' in formErrors
+                    ? (formErrors.city as string)
+                    : '',
+                state:
+                  formErrors && 'state' in formErrors
+                    ? (formErrors.state as string)
+                    : '',
+                country:
+                  formErrors && 'country' in formErrors
+                    ? (formErrors.country as string)
+                    : '',
+                zipCode:
+                  formErrors && 'zipCode' in formErrors
+                    ? (formErrors.zipCode as string)
+                    : '',
+              }}
```

Comment:
Could we align the types here and in other places (for `formErrors`) to avoid the repeated checks and casting?

Reply:
Did you have an idea how to do this? I can look into it more, definitely would be ideal, but since we're group the different validations this is the only way I know to make it not throw an error saying the property doesn't exist.